azure-pipelines.yml

```diff
@@ -63,23 +63,33 @@ stages:
               Debug:
                 _BuildConfig: Debug
         steps:
-        - script: eng\common\cibuild.cmd
-            -configuration $(_BuildConfig)
-            -prepareMachine
-            $(_InternalBuildArgs)
-          name: Build
-          displayName: Build
-          condition: succeeded()
+        - ${{ if eq(variables._RunAsPublic, False) }}:
+          - script: eng\InternalBuild.cmd
+              -configuration $(_BuildConfig)
+              -prepareMachine
+              $(_InternalBuildArgs)
+            name: Build
+            displayName: Build
+            condition: succeeded()
+        - ${{ if eq(variables._RunAsPublic, True) }}:
+          - script: eng\common\CIBuild.cmd
+              -configuration $(_BuildConfig)
+              -prepareMachine
+              $(_InternalBuildArgs)
+            name: Build
+            displayName: Build and test
+            condition: succeeded()
 
-        - task: PublishTestResults@2
-          displayName: 'Publish Unit Test Results'
-          inputs:
-            testResultsFormat: xUnit
-            testResultsFiles: '$(Build.SourcesDirectory)/artifacts/TestResults/**/*.xml'
-            mergeTestResults: true
-            searchFolder: $(system.defaultworkingdirectory)
-            testRunTitle: XHarness unit tests - $(Agent.JobName)
-          condition: succeededOrFailed()
+        - ${{ if eq(variables._RunAsPublic, True) }}:
+          - task: PublishTestResults@2
+            displayName: 'Publish Unit Test Results'
+            inputs:
+              testResultsFormat: xUnit
+              testResultsFiles: '$(Build.SourcesDirectory)/artifacts/TestResults/**/*.xml'
```

Comment:
AFAIK, we have NUnit and xUnit tests atm. We should move https://github.com/dotnet/xharness/blob/master/tests/Microsoft.DotNet.XHarness.iOS.Tests/AppInstallerTests.cs to xUnit. I have created an issue for that: https://github.com/dotnet/xharness/issues/65

Reply:
Good catch, forgot to migrate those as well. Fixed in #68